src/main/java/com/google/cloud/tools/maven/RunAsyncMojo.java

```diff
@@ -0,0 +1,26 @@
+package com.google.cloud.tools.maven;
+
+import org.apache.maven.plugins.annotations.Execute;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Starts running App Engine Development App Server asynchronously.
+ */
+@Mojo(name = "start")
+@Execute(phase = LifecyclePhase.PACKAGE)
+public class RunAsyncMojo extends RunMojo {
+
+  private CountDownLatch waitStartedLatch;
+
+  /**
+   * Configures an asynchronous process runner for running server.
+   */
+  public RunAsyncMojo() {
+    super();
+    processRunner.setAsync(true);
```

Comment:
I think with redirectIO we wont be able to detect that the server is started, I did something like this : https://github.com/GoogleCloudPlatform/app-gradle-plugin/pull/4/files#diff-3bf6c770f2ac694a154614998f389123R60